### PR TITLE
🚚 restructure folders with basic melos

### PR DESCRIPTION
**Pull request by XanderD99**
move base project in seperate folder
prefix all folders with 'flutter_'
update folders where needed in code
update folders in .githubworkflows
add code-workspace file for those who want it 😄

### CODE REVIEW

**Comment by kuhnroyal** [comment]
Awesome, thanks!

> prefix all folders with 'flutter_'

What is the motivation for this? I think it would be preferrable to use the package names.

**Comment by XanderD99** [comment]
Just to have an additional clarification that this is a flutter implementation and not something else. 

But if you prefer the package name as folder name it is a low effort to change it.

File: .github/workflows/flutter_ci.yml [hunk 1]
```diff
@@ -16,8 +16,11 @@ jobs:
       - uses: subosito/flutter-action@v2
         with:
           cache: true
+      - run: flutter pub get
+      - name: melos bootstrap
+        run: dart run melos bootstrap
       - name: Lint analysis
-        run: dart format --set-exit-if-changed .
+        run: dart run melos format --set-exit-if-changed .
```
Comment (josxha) [inline .github/workflows/flutter_ci.yml]: `dart format` runs recursive for all files in `.` so there is no need to use melos here.

File: .github/workflows/flutter_ci.yml [hunk 2]
```diff
@@ -32,8 +35,10 @@ jobs:
          cd scripts
          flutter pub get 
          cd ..
+      - name: melos bootstrap
+        run: dart run melos bootstrap
       - name: Lint analysis
-        run: flutter analyze
+        run: dart run melos analyze
```
Comment (josxha) [inline .github/workflows/flutter_ci.yml]: Same here: `flutter analyze` analyzes all dart files recusively.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -12,14 +12,14 @@ jobs:
...
     uses: ./.github/workflows/publish-single.yml
     with:
-      working-directory: maplibre_gl_web
+      working-directory: flutter_maplibre_gl_web
   publish_maplibre_gl:
     needs: [publish_maplibre_gl_platform_interface, publish_maplibre_gl_web]
     uses: ./.github/workflows/publish-single.yml
     with:
-      working-directory: .
+      working-directory: flutter_maplibre_gl
```
Comment (josxha) [inline .github/workflows/publish.yml]: I'd prefer to have them named like the package name too (without the `flutter_` prefix), sorry. (:

File: .gitignore [hunk 4]
```diff
@@ -22,30 +21,98 @@ migrate_working_dir/
 .settings/
 .vscode/
 
+# Flutter repo-specific
```
Comment (josxha) [inline .gitignore]: What's causes all these new .gitignore entries?

File: .gitignore [hunk 5]
```diff
@@ -22,30 +21,98 @@ migrate_working_dir/
...
+!**/ios/**/default.perspectivev3
+!/packages/flutter_tools/test/data/dart_dependencies_test/**/.packages
+!/dev/ci/**/Gemfile.lock
+google-java-format-1.13.0-all-deps.jar
 
-# Misc
-GeneratedPluginRegistrant.swift
\ No newline at end of file
+# melos
+pubspec_overrides.yaml
```
Comment (josxha) [inline .gitignore:53]: `example/pubspec_overrides.yaml` should be deleted, too.

File: android/gradle/wrapper/gradle-wrapper.jar
Comment (josxha) [inline android/gradle/wrapper/gradle-wrapper.jar:1]: Not sure if we should remove gradle-wrapper.jar? As far as I know projects normally keep it in version control.

File: flutter_maplibre_gl.code-workspace [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
+    "folders": [
+        {
+            "name": "exampel",
```
Comment (josxha) [inline flutter_maplibre_gl.code-workspace]:
```suggestion
            "name": "example",
```

File: flutter_maplibre_gl/CHANGELOG.md
Comment (josxha) [inline flutter_maplibre_gl/CHANGELOG.md:1]: Could be a great opportunity for a symlink.

File: flutter_maplibre_gl/LICENSE
Comment (josxha) [inline flutter_maplibre_gl/LICENSE:1]: Same here, a symlink would be good.

File: flutter_maplibre_gl/analysis_options.yaml [hunk 7]
```diff
@@ -0,0 +1 @@
+include: package:very_good_analysis/analysis_options.yaml
```
Comment (josxha) [inline flutter_maplibre_gl/analysis_options.yaml]:
```suggestion
include: ../analysis_options.yaml
```

This should resolves the failing CI, too.

File: flutter_maplibre_gl_platform_interface/pubspec.yaml [hunk 8]
```diff
@@ -8,10 +8,11 @@ environment:
   sdk: ">=3.0.0 <4.0.0"
   flutter: ">=3.10.0"
 
+dev_dependencies:
+  very_good_analysis: ^5.0.0
+
```
Comment (josxha) [inline flutter_maplibre_gl_platform_interface/pubspec.yaml]: Let's keep the `dev_dependencies` beneath the regular `dependencies`

File: melos.yaml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+name: flutter_maplibre_gl
+
+packages:
+  - example
+  - flutter_maplibre_gl
+  - flutter_maplibre_gl_platform_interface
+  - flutter_maplibre_gl_web
```
Comment (josxha) [inline melos.yaml]:
```suggestion
  - flutter_maplibre_gl_web
  - scripts
```

File: pubspec.yaml [hunk 10]
```diff
@@ -1,34 +1,8 @@
-name: maplibre_gl
-description: A Flutter plugin for integrating MapLibre Maps inside a Flutter application on Android, iOS and web platforms.
-version: 0.19.0+2
-repository: https://github.com/maplibre/flutter-maplibre-gl
-issue_tracker: https://github.com/maplibre/flutter-maplibre-gl/issues
+name: flutter_maplibre_gl
```
Comment (josxha) [inline pubspec.yaml]:
```suggestion
name: maplibre_gl_workspace
repository: https://github.com/maplibre/flutter-maplibre-gl
issue_tracker: https://github.com/maplibre/flutter-maplibre-gl/issues
publish_to: 'none'
```

**Review by josxha — CHANGES_REQUESTED** [review]
Thanks for doing all the refactoring.
Could you check on my comments please?

File: .github/workflows/flutter_ci.yml (hunk 1 above)
Comment (kuhnroyal) [reply .github/workflows/flutter_ci.yml]: I think this is actually better to run on package level, leads to better output and you know which package failed.

File: pubspec.yaml (hunk 10 above)
Comment (kuhnroyal) [reply pubspec.yaml]: I would change the name here to `maplibre_gl_workspace`

File: android/gradle/wrapper/gradle-wrapper.jar
Comment (kuhnroyal) [reply android/gradle/wrapper/gradle-wrapper.jar:1]: Yep, this needs to stay.

File: .github/workflows/flutter_ci.yml [hunk 11]
```diff
@@ -16,8 +16,11 @@ jobs:
       - uses: subosito/flutter-action@v2
         with:
           cache: true
+      - run: flutter pub get
+      - name: melos bootstrap
+        run: dart run melos bootstrap
```
Comment (kuhnroyal) [inline .github/workflows/flutter_ci.yml]: This whole block can be replaced with `https://github.com/bluefireteam/melos-action` and then we can run melos commands like `melos format --set-exit-if-changed .`.

File: .github/workflows/flutter_ci.yml (hunk 1 above)
Comment (josxha) [reply .github/workflows/flutter_ci.yml]: Alright, fine 😄

File: .github/workflows/flutter_ci.yml (hunk 2 above)
Comment (josxha) [reply .github/workflows/flutter_ci.yml]: https://github.com/maplibre/flutter-maplibre-gl/pull/453#discussion_r1630992743

File: pubspec.yaml (hunk 10 above)
Comment (josxha) [reply pubspec.yaml]: Updated the suggestion.

File: .gitignore (hunk 4 above)
Comment (XanderD99) [reply .gitignore]: might be my fault of moving the wrong .gitignore file

**Comment by XanderD99** [comment]
Took a moment but everything should be resolved except for renaming to `maplibre_gl_workspace`. `bluefireteam/melos-action` was complaining that it expected name to be maplibre_gl ¯\_(ツ)_/¯

**Comment by josxha** [comment]
> Took a moment but everything should be resolved except for renaming to maplibre_gl_workspace.

The example app had the wrong path for `maplibre_gl`. Fixed it.

I'll do a rebase and add a CHANGELOG entry. 👍

**Comment by josxha** [comment]
Oof, that was a rebase to remember🤯
I updated the flutter beta workflow to use melos, too. I think that should be all.👍

**Review by josxha — APPROVED** [review]

File: .github/workflows/flutter_beta.yml [hunk 12]
```diff
@@ -25,6 +25,7 @@ jobs:
       with:
         channel: ${{ env.FLUTTER_CHANNEL }}
         cache: true
+    - uses: bluefireteam/melos-action@v3
     - run: flutter pub get
```
Comment (kuhnroyal) [inline .github/workflows/flutter_beta.yml]: I think the pub get here is redundant now since melos does this during bootstrap.
Same in the other jobs.

Comment (josxha) [reply .github/workflows/flutter_beta.yml]: Fixed

**Review by kuhnroyal — APPROVED** [review]
Nice work!

**Pull request merged**